### PR TITLE
Added functionality to pass in github auth_token

### PR DESCRIPTION
**Pull request by dustindall**
Added install_github auth_token as an environment variable. This allows one to install private github repositories on the Azure cluster (I needed this). I would have passed it into the install_github function but there is a bug that does not pass the auth_token to additional remotes which I needed ([#87 under remotes](https://github.com/r-lib/remotes/issues/87)).

### CODE REVIEW

File: R/cluster.R [hunk 1]
```diff
@@ -84,7 +84,8 @@ generateClusterConfig <- function(fileName, ...){
       ),
       rPackages = list(
         cran = vector(),
-        github = vector()
+        github = vector(),
+        authToken = NULL
```
Comment (paselem) [inline R/cluster.R]: Is the authToken only for gitHub? If so, I would rename to githubAuthToken or consider making this a complex object (probably not ideal for backwards compatibility).

@brnleehng - thoughts on this?

Also, we probably should update some docs to help people understand this.

Comment (dustindall) [inline R/cluster.R]: Yes, the token is only for github. Also, I changed authToken to githubAuthToken and pushed it out.

**Comment by dustindall** [comment]
Can we call it githubAuthenticationToken? In the config file under rPackages, github is all lower case and it would be a small nuance to have the H capitalized in one but not the other.

**Comment by paselem** [comment]
Yes, that seems reasonable to me. Thanks for the attention to detail :).

**Comment by dustindall** [comment]
Made the change

**Pull request merged**